src/redux/books/booksSlice.js

```diff
@@ -0,0 +1,46 @@
+import { createSlice, current } from '@reduxjs/toolkit';
+
+const initialState = {
+  bookList: [{
+    id: 0,
+    author: 'Milton',
+    title: 'The lost paradise',
+  },
+  {
+    id: 1,
+    author: 'Cervantes',
+    title: 'The Quijote',
+  },
+  {
+    id: 3,
+    author: 'Moises',
+    title: 'Genesis',
+  },
+  ],
+  id: 2,
+};
+/* eslint no-param-reassign: "error" */
+const bookSlice = createSlice({
+  name: 'book',
+  initialState,
+  reducers: {
+    addBook: (state, { payload }) => {
+      console.log('payload', payload);
+      state.bookList.push(payload);
+      state.id += 1;
+      console.log('id', state.id);
+      console.log('state', current(state.bookList));
+    },
+    removeBook: (state, action) => {
+      const { id } = action.payload;
+      state.bookList = state.bookList.filter((item) => item.id !== id);
+      state.id -= 1;
+      console.log('ID', id);
+      console.log('payload', action.payload);
+      console.log('state', state.bookList);
+    },
+  },
+});
```

Comment:
- [ ] Hi @edahigure, great effort working on your bookstore project, but i notice in your project you have given some data passed in as `array of an object` in your bookSlice. and with this anytime i tried to add a new book to the list, it was only the ones you have in the array that got publish to the list. I can only remove the books you have in the array, but anytime i tried to add a new book, it is the ones you have in the array that got published.
  - I would advise you set the initial state of the book to an empty array. Kindly check the code snippet below for your peruse
  
```JS
// ./src/redux/books/booksSlice.js

import { createSlice } from "@reduxjs/toolkit"

export const booksSlice = createSlice({
  name: "books",
  initialState: {
    books: [],
  },
  reducers: {
    addBook: (state, action) => {
      state.books = [
        ...state.books,
        {
          id: Math.floor(Math.random() * 100),
          title: action.payload.title,
          author: action.payload.author,
        },
      ]
    },
    removeBook: (state, action) => {
      state.books = state.books.filter((book) => book.id !== action.id)
    },
  },
})

export const { addBook, removeBook } = booksSlice.actions

export default booksSlice.reducer
```

- [ ] Also i would suggest not to leave the `console.log` in your work as it pre-dispose your work to the hackers and this practice is not a best practice of writing JavaScript codes. 

Below is the video illustration of my claim: 👇 

[New Recording - 3_2_2023, 5_24_29 PM.webm](https://user-images.githubusercontent.com/78634841/222609355-a0c9dadf-4af4-4aad-b54e-8beb67398e5c.webm)